Source/cmGlobalNinjaGenerator.cxx

```diff
@@ -1140,6 +1141,48 @@ void cmGlobalNinjaGenerator::AddAdditionalCleanFile(std::string fileName,
   this->Configs[config].AdditionalCleanFiles.emplace(std::move(fileName));
 }
 
+void cmGlobalNinjaGenerator::AddLinkCommand(
+  const std::string& linkCommand, std::vector<std::string> objectFiles,
+  const std::vector<std::string>& librariesFiles, const std::string& linkCommandsVersion)
+{
+  // Compute Ninja's build file path.
+  std::string buildFileDir =
+    this->GetCMakeInstance()->GetHomeOutputDirectory();
+
+  for(auto & objectFile : objectFiles){
+    objectFile = cmStrCat(buildFileDir, "/", objectFile);
+  }
+
+  objectFiles.insert(objectFiles.end(), librariesFiles.begin(), librariesFiles.end());
```

Comment:
librariesFiles is only for static libraries or dynamic libraries also?

Reply:
dynamic and static